commit b8ce25df2999ac6a135ce1bd14b7243030a1338a
Author: David Kaplan <david.kaplan@amd.com>
Date:   Wed Jan 8 14:24:43 2025 -0600

    x86/bugs: Add AUTO mitigations for mds/taa/mmio/rfds
    
    Add AUTO mitigations for mds/taa/mmio/rfds to create consistent vulnerability
    handling.  These AUTO mitigations will be turned into the appropriate default
    mitigations in the <vuln>_select_mitigation() functions.  Later, these will be
    used with the new attack vector controls to help select appropriate
    mitigations.
    
    Signed-off-by: David Kaplan <david.kaplan@amd.com>
    Signed-off-by: Borislav Petkov (AMD) <bp@alien8.de>
    Link: https://lore.kernel.org/r/20250108202515.385902-4-david.kaplan@amd.com

diff --git a/arch/x86/include/asm/processor.h b/arch/x86/include/asm/processor.h
index c0cd10182e90..90278d0c071b 100644
--- a/arch/x86/include/asm/processor.h
+++ b/arch/x86/include/asm/processor.h
@@ -757,6 +757,7 @@ extern enum l1tf_mitigations l1tf_mitigation;
 
 enum mds_mitigations {
 	MDS_MITIGATION_OFF,
+	MDS_MITIGATION_AUTO,
 	MDS_MITIGATION_FULL,
 	MDS_MITIGATION_VMWERV,
 };